framework/src/org/checkerframework/framework/util/element/SuperTypeApplier.java

```diff
@@ -73,15 +73,16 @@ public static void annotateSupers(
      */
     @Override
     public int getElementIndex() {
-        return index;
+        return index == 0xffff ? -1 : index;
     }
 
     /**
      * @return the type_index of anno's TypeAnnotationPosition
      */
     @Override
     public int getTypeCompoundIndex(Attribute.TypeCompound anno) {
-        return anno.getPosition().type_index;
+        int type_index = anno.getPosition().type_index;
+        return type_index == 0xffff ? -1 : type_index;
```

Comment:
Can you add a TODO comment that this is a temporary workaround until the corresponding langtools method is fixed.